specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2018-06-01/QueryPerformanceInsights.json

```diff
@@ -75,6 +75,305 @@
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/topQueryStatistics/{queryStatisticId}": {
+      "get": {
+        "tags": [
+          "TopQueryStatistics"
+        ],
+        "operationId": "TopQueryStatistics_Get",
+        "x-ms-examples": {
+          "TopQueryStatisticsGet": {
+            "$ref": "./examples/TopQueryStatisticsGet.json"
+          }
+        },
+        "description": "Retrieve the query statistic for specified identifier.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/QueryStatisticIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/QueryStatistic"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/topQueryStatistics": {
+      "get": {
+        "tags": [
+          "TopQueryStatistics"
+        ],
+        "operationId": "TopQueryStatistics_ListByServer",
+        "x-ms-examples": {
+          "TopQueryStatisticsListByServer": {
+            "$ref": "./examples/TopQueryStatisticsListByServer.json"
+          }
+        },
+        "description": "Retrieve the Query-Store top queries for specified metric and aggregation.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
```

Comment:
GET requests can not have a body

Reply:
This is an existing API and cannot be changed, similar to [this](https://github.com/Azure/azure-rest-api-specs/blob/412750c48965135973454b582a81901c10529d5b/specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2018-06-01/QueryPerformanceInsights.json#L87)

Reply:
Please point to an example GET request in the ARM logs that sent a body.  For the example you linked I see all those parameters sent in the query string on all calls to that API, not the body.

Example:  https://management.azure.com:443/subscriptions/.../resourceGroups/.../providers/Microsoft.DBforPostgreSQL/servers/.../topQueryStatistics?api-version=2018-06-01&numberOfTopQueries=5&aggregationFunction=avg&observedMetric=duration&observationStartTime=2021-01-27T21:15:00.000Z&observationEndTime=2021-01-28T21:30:00.000Z&aggregationWindow=PT15M

https://armprod.kusto.windows.net/ARMProd?query=H4sIAAAAAAAEAPMoKSnwzEvOz83MSw9KLSxNLS4p5qpRKM9ILUpVCPH0dQ0OcfQNULBTSEzP1zDK0OSFS5YkFqWnlgSlFueXFiWnhlQWpCrY1ikoFacWlaUWFeuX5BcATSuqLC5JLMksLslMLlbi5eLlAgDsO%2f3McAAAAA%3d%3d&web=0

HttpIncomingRequests
| where TIMESTAMP > ago(2h)
| where targetResourceType =~ "servers/topquerystatistics"

Reply:
Checked the code implementation and it does look like we don't read anything from the body parameters, thanks for catching this!